mule-deployer/src/main/java/org/mule/tools/client/core/AbstractClient.java

```diff
@@ -128,4 +145,33 @@ protected void configureRequest(Invocation.Builder builder) {
 
   }
 
+  protected void configureSecurityContext(ClientBuilder builder) {
+    // Implemented in concrete classes
+  }
+
+  protected void checkResponseStatus(Response response) {
+    if (familyOf(response.getStatus()) != SUCCESSFUL) {
+      throw new ClientException(response);
+    }
+  }
+
+  protected void checkResponseStatus(Response response, Status... successStatus) {
```

Comment:
I guess the successStatus should be something like expectedStatus or status... I've seen code in other classes where this function was invoked with NOT_FOUND, which is not a sucessStatus. It can be somehow misleading when someone uses this method. You can use the plural statuses if you want.

Reply:
yeah this is related to the other one. 
We are following the logic checkeResponseStatus( *, tellMeWhatASuccessIsForYou) 
You see, yes 404 is not by default a success but for the guy using an API it may be that he was aiming for a 404 so for him that's a success. 

That's the semantic behind it but tell you what I will change this one.

Reply:
:joy: